### PR TITLE
Occlum's runtime-boot

### PR DESCRIPTION
**Pull request by mythi**

### CODE REVIEW

**Comment by mythi** [comment]
> Some updates are required when Occlum v1.0 released.

my idea is that we can move the development/maintanence of this functionality to this repository. we can commit the necessary changes once the build tools move to use the v1.0 Occlum.

**Comment by qzheng527** [comment]
> > Some updates are required when Occlum v1.0 released.
> 
> my idea is that we can move the development/maintanence of this functionality to this repository. we can commit the necessary changes once the build tools move to use the v1.0 Occlum.

Absolutely.

File: src/runtime-boot/init.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+includes:
+  - base.yaml
+targets:
+  - target: /bin/
+    copy:
+      - files:
+        - ../init
```
[the remark this answers is not in the document]
Comment (qzheng527) [reply src/runtime-boot/init.yaml:7]: @hairongchen-intel The source path in yaml file is relative to the working dir to run copy_bom tool.
For detail you could refer to the original [`runtime-boot`](https://github.com/occlum/occlum/tree/enclave-cc/demos/runtime_boot)

[the remark this answers is not in the document]
Comment (hairongchen-intel) [reply src/runtime-boot/init.yaml:7]: the relative path makes sense

**Comment by mythi** [comment]
as part of #45, let's get this merged so that the code can be built into `boot_instance` from this repo.

**Comment by qzheng527** [comment]
LGTM.
btw, I will add runtime env support soon. We may need update then.

**Comment by mythi** [comment]
> LGTM.
> 
> btw, I will add runtime env support soon. We may need update then.

Would it be possible to open a PR in this repo for those changes?

**Review by dcmiddle — APPROVED** [review]

**Comment by jiere** [comment]
> > LGTM.
> > btw, I will add runtime env support soon. We may need update then.
> 
> Would it be possible to open a PR in this repo for those changes?

IMO, if those occlum change is only for enclave-cc, it makes sense to merge in enclave-cc repo. Otherwise, double commit may be needed.
Anyway, Mikko, just merge the current PR first.

**Comment by qzheng527** [comment]
> > LGTM.
> > btw, I will add runtime env support soon. We may need update then.
> 
> Would it be possible to open a PR in this repo for those changes?

Sure. 
And not only the changes here, but also the occlum bundle requires change as well due to the libos got changed.

**Comment by mythi** [comment]
> if those occlum change is only for enclave-cc, it makes sense to merge in enclave-cc repo.

Any new features in the libOS itself we would consume through the Occlum releases. This `init` is a custom program for enclave-cc and we will move to maintain in this repo.

**Comment by mythi** [comment]
> Anyway, Mikko, just merge the current PR first.

ack

**Pull request merged**